cmd/lifecycle-util.go

```diff
@@ -0,0 +1,277 @@
+/*
+ * MinIO Client (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"errors"
+	"math"
+	"net/url"
+	"os"
+	"strings"
+
+	stdlibjson "encoding/json"
+	"encoding/xml"
+
+	"github.com/mattn/go-isatty"
+	json "github.com/minio/mc/pkg/colorjson"
+
+	"github.com/minio/mc/pkg/probe"
+	minio "github.com/minio/minio-go/v6"
+	"github.com/minio/minio-go/v6/pkg/credentials"
+	"github.com/minio/minio/pkg/console"
+)
+
+const (
+	idWidth             int = 12
+	prefixWidth         int = 10
+	statusWidth         int = 12
+	expiryWidth         int = 8
+	expiryDatesWidth    int = 14
+	tagWidth            int = 22
+	transitionWidth     int = 14
+	transitionDateWidth int = 18
+	storageClassWidth   int = 18
+)
+
+const (
+	leftAlign   int = 1
+	centerAlign int = 2
+	rightAlign  int = 3
+)
+
+const (
+	idLabel             string = "ID"
+	prefixLabel         string = "Prefix"
+	statusLabel         string = "Enabled "
+	statusDisabledLabel string = "Disabled"
+	expiryLabel         string = "Expiry"
+	expiryDatesLabel    string = "Date/Days "
+	singleTagLabel      string = "Tag"
+	tagLabel            string = "Tags"
+	transitionLabel     string = "Transition"
+	transitionDateLabel string = "Date/Days "
+	storageClassLabel   string = "Storage-Class "
+	forceLabel          string = "force"
+	allLabel            string = "all"
+)
+
+const (
+	statusLabelKey          string = "Enabled"
+	storageClassLabelKey    string = "Storage-Class"
+	expiryDatesLabelKey     string = "Expiry-Dates"
+	transitionDatesLabelKey string = "Transition-Date"
+	transitionDaysLabelKey  string = "Transition-Days"
+)
+
+const (
+	expiryDatesLabelFlag string = "Expiry-Date"
+	expiryDaysLabelFlag  string = "Expiry-Days"
+)
+
+const (
```

Comment:
all of them too generic names, prefix it with ILM to avoid polluting the global namespace